_posts/2020-08-01-Making-IrishGen-SPARQL-Part-III.md

```diff
@@ -0,0 +1,144 @@
+---
+layout: post
+date: 2020-08-01
+title: "Making IrishGen SPARQL Part III: Ask and Describe"
+author: cgy
+---
+
+{% assign author = site.data.authors[page.author]%}
+{{author.name}}
+
+ORCID: <a href="https://orcid.org/{{ author.orcid }}" title="{{author.name}}">{{author.orcid}}</a>
+
+Over the last two posts ([part I]({% post_url
+2020-06-26-Making-IrishGen-SPARQL-Part-I %}), [part II]({% post_url
+2020-08-05-Making-IrishGen-SPARQL-Part-II %})), the two main query
+forms of SPARQL (Select and Construct) have been shown with examples
+from IrishGen.  These two query forms together comprise the bulk of
+the operations a regular user of SPARQL will need to use on a regular
+basis.  This post will explain the uses of the two less often used
+query forms:
+[Ask](https://www.w3.org/TR/2013/REC-sparql11-query-20130321/#ask) and
+[Describe](https://www.w3.org/TR/2013/REC-sparql11-query-20130321/#describe).
+These will be covered in the order in which they are most often used.
+
+# Describe
+
+The Describe query will return the RDF about a single URL supplied by
+the user.  What is returned by a Triplestore is undefined by SPARQL
+(and deliberately so) in the standard.  What is generally returned is
+the RDF that the Triplestore has about a particular URL.  This is
+useful since, if the user knows the URL, they can interrogate the
+Triplestore about what is stored in the Triplestore about that URL
+without needing to go through the ceremony of a construct query.
+Moreover, Describe queries are often used in visual representations of
+RDF graphs.
+
+For instance, the screenshot below is from GraphDB's "Visual Graph"
+feature and uses Finn mac Cumaill as the exemplar.  What happens is
+that the user sends the URL in which they are interested, GraphDB then
+makes a Describe query on the URL, and RDF returned is used by GraphDB
+to create a visual representation.  This is very useful when
+investigating single individuals within IrishGen.  Additionally, it is
+good if a user wants to have a secure, single starting point to
+explore the graph in various ways.
+
+<img src="{{site.baseurl}}/assets/images/describe_part3.png" />
+
+Using the "Expand" feature of GraphDB's Visual Graph will send further
+Describe queries to the Triplestore.  The resulting RDF will then be
+rendered in visual form and any connections will be presented to the
+user.  Describe queries can also be done as plain SPARQL as well.  For
+instance:
+
+```sparql
+describe <http://example.com/LL/lagin.trig#Find>
+```
+
+will return over 1000 results giving everything that the Triplestore
+contains about Find.  The result of this is that the Visual Graph can
```

Comment:
Not just any Find.  This is still Finn mac Cumaill, yes? Perhaps say so, as the more archaic spelling of his name might imply this is a different person?